File: typescript/tmp5/main.ts

function main() {
    const nums = [1, 2, 3, 4];
    const target = 7;
    let resp = run(nums, target);
    console.log(resp)
}

function run(nums: number[], target: number): [number, number] {
    if (nums.length < 2) return [];

    const seen = new Map<number, number>();

    for (let i = 0; i < nums.length; i++) {
        const need = target - nums[i];
    }
}


main()

File: cpp/tmp5/main.cpp
#include <iostream>
#include <unordered_map>
#include <vector>

std::pair<int, int> twoSum(const std::vector<int>& nums, int target) {
    if (nums.size() < 2) {
        return {-1, -1};
    }

    std::unordered_map<int, int> seen;

    for (int i = 0; i < nums.size(); i++) {
        int num = nums[i];
        int complement = target - num;

        if (seen.find(complement) != seen.end()) {
            int compIdx = seen[complement];
            return {compIdx, i};
        }

        seen[num] = i;
    }

    return {-1, -1};
}

int main() {
    std::vector<int> nums = {1, 2, 3, 4};
    int target = 7;
    
    auto result = twoSum(nums, target);

    if (result.first != -1) {
        std::cout << result.first << result.second << std::endl; 
    } else {
        std::cout << "False" << std::endl;
    }

    return 0;
}
